File: url-shortening-service/src/main/java/com/project/url_shortening_service/repository/UserRepository.java
package com.project.url_shortening_service.repository;

import com.project.url_shortening_service.model.UrlMapping;
import org.springframework.stereotype.Repository;

import java.util.concurrent.ConcurrentHashMap;

@Repository
public class UserRepository {
    private final ConcurrentHashMap<String, UrlMapping> db = new ConcurrentHashMap<>();

    public void addUrlMapping(UrlMapping urlMapping) {
        db.put(urlMapping.getId(), urlMapping);
    }

    public UrlMapping getUrlMapping(String shortUrl) {
        return db.getOrDefault(shortUrl, null);
    }
}

File: url_shortening_service_flask/repositories/user_repository.py
class UserRepository:
    def __init__(self):
        self.db = {}

    def add_url_mapping(self, url_mapping):
        self.db[url_mapping.id] = url_mapping.to_dict()
        print(self.db[url_mapping.id])

    def get_url_mapping(self, short_url):
        return self.db.get(short_url)
